dateparser/parser.py

```diff
@@ -161,11 +155,11 @@ def _find_best_matching_date(cls, datestring):
     @classmethod
     def parse(cls, datestring, settings):
         if not no_space_parser_eligibile(datestring):
-            return
+            raise ValueError('Unable to parse date from: %s' % datestring)
 
         datestring = datestring.replace(':', '')
         if not datestring:
-            return
+            raise ValueError("Empty string")
```

Comment:
Does this change the current behaviour? Is it a breaking change?

Reply:
As explained, in the PR description: "I made some changes in the tests as it was expected to return None when finding invalid inputs but it should return an empty date_data ({'date_obj': None, 'period': ...}) or raising an exception and I preferred the second approach". This decision was made to make it behave more similar to the `absolute-time` parser.

About the "breaking change", this PR has been tagged as it. The full implementation is a breaking change.